File: onto-code-maker-result-tester/src/main/java/ontology/generator/classes/examples/family/serialization/PetSerialization.java

package ontology.generator.classes.examples.family.serialization;

import org.eclipse.rdf4j.model.*;
import java.util.Collection;
import java.util.Set;
import java.util.HashSet;
import java.util.*;

import java.util.stream.Collectors;
import ontology.generator.classes.examples.family.entities.*;
import org.eclipse.rdf4j.model.util.Values;
import org.eclipse.rdf4j.model.vocabulary.RDF;

import java.time.LocalTime;
import ontology.generator.classes.examples.family.Vocabulary;
import ontology.generator.classes.examples.family.entities.Pet;

public class PetSerialization extends SerializationModel<Pet>{

    @Override
    public void addToModel(Model model, Pet pet) {
        model.add(pet.getIri(),RDF.TYPE, pet.getClassIRI());

    }


    protected void setProperties(Model model,Pet pet,int nestingLevel) throws Exception{

    }

    @Override
    public Pet getInstanceFromModel(Model model,IRI instanceIri,int nestingLevel) throws Exception{
        Model statements = model.filter(instanceIri,RDF.TYPE,Pet.CLASS_IRI);
        if(statements.size() != 0){
            Pet pet = new Pet(instanceIri);
            if(nestingLevel > 0){
                nestingLevel--;
                setProperties(model, pet,nestingLevel);
            }
            return pet;
        }

        return null;
    }

    @Override
    public Collection<Pet> getAllInstancesFromModel(Model model,int nestingLevel)throws Exception{
        Model statements = model.filter(null,RDF.TYPE,Pet.CLASS_IRI);
        Collection<Pet> allInstances = new ArrayList<>();
        for(Statement statement:statements){
            Resource subject = statement.getSubject();
            if(subject.isIRI()){
                IRI iri = (IRI) subject;
                Pet pet = getInstanceFromModel(model,iri,nestingLevel);
                allInstances.add(pet);
            }
        }

        return allInstances;
    }

    @Override
    public void removeInstanceFromModel(Model model,IRI instanceIri) {
        Set<Value> rdfCollections = model.filter(instanceIri,null,null).objects();
        List<Value> listOfnodes = rdfCollections.stream().filter(o -> o.isBNode()).collect(Collectors.toList());
        for(Value node:listOfnodes){
            model.removeAll(getModelRDFCollection(model,(BNode)node));
            model.remove(instanceIri,null,(BNode)node);
        }
        model.remove(instanceIri,RDF.TYPE,Pet.CLASS_IRI);
        model.remove(instanceIri,null,null);
    }

    @Override
    public void updateInstanceInModel(Model model,Pet pet){
        Set<Value> rdfCollections = model.filter(pet.getIri(),null,null).objects();
        List<Value> listOfnodes = rdfCollections.stream().filter(Value::isBNode).collect(Collectors.toList());
        for(Value node:listOfnodes){
            model.removeAll(getModelRDFCollection(model,(BNode)node));
            model.remove(pet.getIri(),null,node);
        }

        Model statements = model.filter(pet.getIri(),null,null);
        statements.removeIf(x -> !x.getPredicate().equals(RDF.TYPE));

    }



}

File: onto-code-maker-result-tester/src/test/python/familyPython/serialization/PetSerialization.py

from rdflib.graph import URIRef, Literal,BNode,Collection
from rdflib.namespace import RDF
from Vocabulary import Vocabulary
from entities.Pet import Pet
from .SerializationModel import SerializationModel

class PetSerialization(SerializationModel):

    def addToModel(self,ontology, pet):
        ontology.add((pet.getIri(),RDF.type, pet.getClassIRI()))



    def getInstanceFromModel(self,ontology, instanceIri,nestingLevel):
        if (instanceIri,RDF.type,Pet.CLASS_IRI) in ontology:
            pet =  Pet(instanceIri)
            if nestingLevel > 0:
                nestingLevel -= 1
            return pet
        return None

    def getAllInstancesFromModel(self,ontology,nestingLevel):
        allInstances = set()
        for s, p, o in ontology.triples((None,RDF.type,Pet.CLASS_IRI)):
            if isinstance(s,URIRef) :
                pet = self.getInstanceFromModel(ontology,s,nestingLevel)
                allInstances.add(pet)
        return allInstances

    def removeInstanceFromModel(self,ontology, subject):
        ontology.remove((subject,RDF.type,Pet.CLASS_IRI))
        for statement in ontology.triples((subject,None,None)):
            ontology.remove(statement)



